### PR TITLE
Enable to align XML in Json2XmlValidator

### PR DESCRIPTION
**Pull request by gvanbrakel**
This PR allows XML to be aligned (elements put to the correct order, according to the XSD) rather than only validated. It also allows values to be substituted from parameters, like for JSON input

### CODE REVIEW

File: core/src/test/java/nl/nn/adapterframework/align/TestAlignXml.java [hunk 1]
```diff
@@ -118,6 +118,16 @@ public void testAbcExtraElementsIgnored() throws Exception {
 		testXml(xmlString, schemaUrl, expectedFailureReason, "extra elements not allowed", false, true);
 	}
 
+	@Test
+	public void testAbcWrongOrder() throws Exception {
+		String schemaFile = "Abc/abc.xsd";
+		String inputFile = "Abc/abc-wrongorder";
+		String expectedFailureReason = null;
+		URL schemaUrl = Utils.class.getResource(BASEDIR + schemaFile);
```
Comment (nielsm5) [inline core/src/test/java/nl/nn/adapterframework/align/TestAlignXml.java:126]:
```suggestion
		URL schemaUrl = TestFileUtils.getTestFileURL(BASEDIR + schemaFile);
```

File: core/src/main/java/nl/nn/adapterframework/pipes/Json2XmlValidator.java [hunk 2]
```diff
@@ -167,7 +171,18 @@ public PipeRunResult doPipe(Message input, PipeLineSession session, boolean resp
 					messageToValidate=addNamespace(messageToValidate); // TODO: do this via a filter
 					//log.debug("added namespace to message [{}]", messageToValidate);
 				}
+
+				// Align XML if required
 				storeInputFormat(DocumentFormat.XML, input, session, responseMode);
+				if (isAlignXml() || getParameterList()!=null && !getParameterList().isEmpty()) {
```
[the remark this answers is not in the document]
Comment (nielsm5) [reply core/src/main/java/nl/nn/adapterframework/pipes/Json2XmlValidator.java:177]: Dit lijkt mij handig!

File: core/src/main/java/nl/nn/adapterframework/align/DomTreeAligner.java [hunk 3]
```diff
@@ -65,43 +65,37 @@ public String getNodeText(XSElementDeclaration elementDeclaration, Node node) {
...
+		}
 		return children;
 	}
 
 	@Override
 	public Iterable<Node> getNodeChildrenByName(Node node, XSElementDeclaration childElementDeclaration) throws SAXException {
 		String name=childElementDeclaration.getName();
-		if (log.isTraceEnabled()) log.trace("getChildrenByName() node ["+node+"] name ["+name+"]");
+		if (log.isTraceEnabled()) log.trace("getNodeChildrenByName() node ["+node+"] name ["+name+"]");
 		List<Node> children = new LinkedList<Node>();
```
[the remark this answers is not in the document]
Comment (philipsens) [reply core/src/main/java/nl/nn/adapterframework/align/DomTreeAligner.java]: The bot is retired

File: core/src/main/java/nl/nn/adapterframework/pipes/Json2XmlValidator.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (philipsens) [reply core/src/main/java/nl/nn/adapterframework/pipes/Json2XmlValidator.java]: The bot is retired

**Comment by philipsens** [comment]
The pull-request is more than a year old and is considered stale, implementation should be revised.

**Pull request closed without merging**